lib/midcom/helper/imagepopup/handler/upload.php

```diff
@@ -1,161 +1,169 @@
+	   }
 	
-	public function getModifyFilename($filename)
-	{
-		$pieces = explode('.', $filename);
-		$core = "modified";
-		$extension = array_pop($pieces);
-		$time = $this->getCurrentTime('d-m-Y-H-i-s');
-		$modifyFilename =  $core . $time . "." . $extension;
+    private function get_modify_filename($filename)
+	   {
+		      $pieces = explode('.', $filename);
+		      $core = "modified";
+		      $extension = array_pop($pieces);
+		      $time = $this->get_current_time('d-m-Y-H-i-s');
+		      $modifyFilename =  $core . $time . "." . $extension;
 	
-		return $modifyFilename;
-	}
+		      return $modifyFilename;
+    }
 	
-	public function getCurrentTime($pattern)
-	{
-		return date($pattern, time());
-	}
+    private function get_current_time($pattern)
+	   {
+		      return date($pattern, time());
+	   }
 	
-	public function getDataFromDatabase($filename, $data)
-	{
-		$query = midcom_db_attachment::new_query_builder();
-		$query->add_constraint('name', '=', $filename);
-		$entry = $query->execute();
-		$guid = $entry[0]->guid;
-		
-		if($data == "guid") return $entry[0]->guid;
-		else if ($data == "target") return $entry[0];
-		
-		return false;
-	}
-	
-	public function writeTheFile($tmp, $target)
-	{
-		$source  = fopen($tmp, 'r');
-		if(!$source)
-		{
-			debug_add("Could not open file '{$this->_filename}' for reading", MIDCOM_LOG_ERROR);
-			return false;
-		}
-		$stat = $target->copy_from_handle($source);
-		if (!$stat)
-		{
-			debug_add("copy_from_handle() failed", MIDCOM_LOG_ERROR);
-		}
-		fclose($source);
-		
-		return $stat;
-	}
-	
-	public function insertDatabase($filename, $mimetype, $parentguid)
-	{
-		$attachment = new midcom_db_attachment;
-		$attachment->name = $filename;
-		$attachment->title = $filename;
-		$attachment->mimetype = $mimetype;
-		$attachment->parentguid = $parentguid;
-		if(!$attachment->create())
-		{
-			throw new midcom_error('Failed to create derived image: ' . midcom_connection::get_error_string());
-			return false;
-		}
-		
-		return true;
-	}
-}
+	   private function get_data_from_database($filename, $parentguid, $data)
+	   {
+		      $query = midcom_db_attachment::new_query_builder();
+		      $query->add_constraint('name', '=', $filename);
+		      $query->add_constraint('parentguid', '=', $parentguid);
+		      $entry = $query->execute();
+		      $guid = $entry[0]->guid;
+		
+		      if($data == "guid") 
+		      {
+		          return $entry[0]->guid;
+		      }
+		      else if($data == "target") 
+		      {
+		          return $entry[0];
+		      }
+		      else 
+		      {
+		          throw new midcom_error('Failed to get data from database: ' . midcom_connection::get_error_string());
+		      } 
+    }
 
-?>
\ No newline at end of file
+    private function write_the_file($tmp, $target)
+	   {
+		      $source  = fopen($tmp, 'r');
+		      if(!$source)
+		      {
+			         debug_add("Could not open file ". $tmp ." for reading", MIDCOM_LOG_ERROR);
```

Comment:
The exception will write to the error log already, there is no need for an addtional `debug_add()` call here